services/requester/batch_tx_pool.go

```diff
@@ -266,8 +306,32 @@ func (t *BatchTxPool) submitSingleTransaction(
 	}
 
 	if err := t.client.SendTransaction(ctx, *flowTx); err != nil {
+		// If there was any error while sending the transaction,
+		// we record it as a dropped transaction.
+		t.collector.TransactionsDropped(1)
 		return err
 	}
 
 	return nil
 }
+
+func (t *BatchTxPool) updateEOAActivityMetadata(
+	from gethCommon.Address,
+	nonce uint64,
+) {
+	t.txMux.Lock()
+	defer t.txMux.Unlock()
```

Comment:
the `eoaActivity` object is still read outside of the lock in `Add()`. I think we need to add the mutex to `eoaActivityMetadata` and aquire it each time we need to interact with the object.